Do not permit value transfers with external calls on Solana (#901)
hyperledger-labs/solang: docs/targets/solana.rst

are `some examples <https://solana-labs.github.io/solana-solidity.js/>`_. There is also
`solang's integration tests <https://github.com/hyperledger-labs/solang/tree/main/integration/solana>`_.

.. _value_transfer:

Transfering native value with a function call
_____________________________________________

The Solidity langauge on Ethereum allows value transfers with an external call.

> Solana Cross Program Invocation (CPI) does not support this. So, this is not
> supported. This means that:

Reviewer: ```suggestion
Solana Cross Program Invocation (CPI) does not support this. This means that:
```

 - Specifying `value:`` on an external call is not permitted
 - The `payable` keyword has no effect
 - `msg.value` is not supported

.. note::

    Hypothetically, the caller could transfer native balance and then inform the
    callee about the amount transferred by specifying this in the instruction data.

>     However, it would be trivial to forge this.

Reviewer: ```suggestion
    However, it would be trivial to forge such an operation.
```
You are using `this` too much!

Receive function
________________

> In Solidity, if defined, the `receive()` function is called, whenever the native
> balance for an account gets credited. On Solana, there is no method to implement
> this. The balance of an accounts can be credited without any code being executed.

Reviewer: ```suggestion
In Solidity, the `receive()` function, when defined, is called whenever the native
balance for an account gets credited. On Solana, there is no method that implement
this. The balance of an account can be credited without any code being executed.
```

`solang's integration tests <https://github.com/hyperledger-labs/solang/tree/main/integration/solana>`_.

.. _value_transfer:

Transfering native value with a function call
_____________________________________________

The Solidity langauge on Ethereum allows value transfers with an external call. Solana
Cross Program Invocation (CPI) does not support this. Hypothetically, the caller
could transfer native balance and then inform the callee about the amount transferred
by specifying this in the instruction data. However, it would be trivial to forge

> this. So, this is not supported at all. This means that:

Reviewer: You repeated `this` three times in these two lines. This sounds awkward.

Author: Rewritten

 - Specifying `value:`` on an external call is not permitted
 - The `payable` keyword has no effect
 - `msg.value` is not supported

Receive function
________________

> In Solidity, a `receive()` function can be defined which gets called, whenever the

Reviewer: `function can be defined which gets called`
This does not sound right.

Author: Rewritten
